agent/consul/leader.go

```diff
@@ -213,17 +230,21 @@ WAIT:
 	for {
 		select {
 		case <-stopCh:
-			return
+			return nil
 		case <-s.shutdownCh:
-			return
+			return nil
 		case <-interval:
 			goto RECONCILE
 		case member := <-reconcileCh:
 			s.reconcileMember(member)
 		case index := <-s.tombstoneGC.ExpireCh():
 			go s.reapTombstones(index)
 		case errCh := <-s.reassertLeaderCh:
-			errCh <- reassert()
+			err := reassert()
+			errCh <- err
```

Comment:
How come we're sending an err into the errCh here? Seems like we should just return if `err != nil` if we don't want to retry the reassert. (since this is the only place where we read from that channel)

Reply:
This channel is also read by the snapshot endpoint. It waits for reassertion to be successful before reporting that back to the caller. This is why we need to send our errors (or `nil`) there.